sdk/core/azure-core-metrics-opentelemetry/src/main/java/com/azure/core/metrics/opentelemetry/OpenTelemetryUtils.java

```diff
@@ -51,17 +49,19 @@ static void addAttribute(io.opentelemetry.api.common.AttributesBuilder attribute
      * If not context is found, returns {@link io.opentelemetry.context.Context#current()}.
      */
     static io.opentelemetry.context.Context getTraceContextOrCurrent(Context azContext) {
-        Optional<Object> traceContextOpt = azContext.getData(PARENT_TRACE_CONTEXT_KEY);
-        if (traceContextOpt.isPresent()) {
-            Object traceContextObj = traceContextOpt.get();
-            if (traceContextObj instanceof io.opentelemetry.context.Context) {
+        Object traceContextObj = azContext.getData(PARENT_TRACE_CONTEXT_KEY).orElse(null);
+        if (traceContextObj != null) {
+            if (io.opentelemetry.context.Context.class.isAssignableFrom(traceContextObj.getClass())) {
                 return (io.opentelemetry.context.Context) traceContextObj;
-            } else if (traceContextObj != null) {
-                // TODO (limolkova) somehow we can get shaded otel agent context here
-                if (!warnedOnContextType) {
-                    LOGGER.warning("Expected instance of `io.opentelemetry.context.Context` under `PARENT_TRACE_CONTEXT_KEY`, but got {}, ignoring it.", traceContextObj.getClass().getName());
-                    warnedOnContextType = true;
-                }
+            } else if (!warnedOnContextType) {
+                // TODO (limolkova) The context we have here is created by azure-core-tracing-opentelemetry.
```

Comment:
Do we have a Github issue to track this?

Reply:
now we do :) https://github.com/Azure/azure-sdk-for-java/issues/36537